### DISCUSSION TITLE
Allow toggling on/off without confirmation

### DISCUSSION THREAD

**Discussion by dim882** [community/community#48594] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

There are situations where Copilot is more of a distraction than anything, and I just want to turn it off. Then I may want to try it in a new situation and turn it back on. I know that it's possible to enable/disable it with shift-cmd-P, then search for the option to configure enabled/disabled, but instead of just toggling it, that opens a dialog that requires a choice confirmation. The dialog is not given focus automatically, so you have to choose with a mouse click. For vim users and people who like to do everything with key commands, this is annoying. Would it be a bad idea to have a choice to toggle it globally without any confirmation? Or am I missing something and this is already possible? Thanks!

**Comment by rileyblackwell** [comment]
I have the same problem. I think a toggle on/off for the current workspace without requiring a confirmation would be a valuable feature.

**Comment by Yuki-cpp** [comment]
Hello
Maybe a bit late, but the following mappings have done the job for me:
```
local copilot_mappings = {
    C = {
        name = "Copilot",
        e = { "<cmd>let g:copilot_enabled = 1<CR>", "Enable copilot" },
        d = { "<cmd>let g:copilot_enabled = 0<CR>", "Disable copilot" },
        p = { "<cmd>Copilot panel<CR>", "Open copilot panel" },
        o = { "<cmd>Copilot open<CR>", "Open copilot in a new window" },
    }
}
wk.register(copilot_mappings, normal_opts)
```

I'm using which-key so there is a bit of boilerplate but you should be able to adapt them easily enough

**Reply by dim882** [reply]
Oh, cool. I didn't realize that was possible. Thanks!

**Reply by dim882** [reply]
Hey @Yuki-cpp, I finally got around to trying to use your snippet, but I realized I'm not really understanding how it works. How does `let g:copilot_enabled = 0` disable copilot? Isn't that just a variable you're creating? If so, how would Copilot use it? I looked through which-key docs and I'm not really getting it. Any clarification you could provide would be appreciated!

**Comment by deepbajud** [comment]
I have same problem @d-imal  your answer solve my problem properly
